injected/src/features/password-import.js

```diff
@@ -0,0 +1,233 @@
+import ContentFeature from '../content-feature'
+import { DDGProxy, DDGReflect, withExponentialBackoff } from '../utils'
+
+const ANIMATION_DURATION_MS = 1000
+const ANIMATION_ITERATIONS = Infinity
+
+/**
+ * This feature is responsible for animating some buttons passwords.google.com,
+ * during a password import flow. The overall approach is:
+ * 1. Check if the path is supported,
+ * 2. Find the element to animate based on the path - using structural selectors first and then fallback to label texts),
+ * 3. Animate the element, or tap it if it should be autotapped.
+ */
+export default class PasswordImport extends ContentFeature {
+    #exportButtonSettings
+    #settingsButtonSettings
+    #signInButtonSettings
+
+    /**
+     * Takes a path and returns the element and style to animate.
+     * @param {string} path
+     * @returns {Promise<{element: HTMLElement|Element, style: any, shouldTap: boolean}|null>}
+     */
+    async getElementAndStyleFromPath (path) {
+        if (path === '/') {
+            const element = await this.findSettingsElement()
+            return element != null
+                ? {
+                    style: {
+                        scale: 1,
+                        backgroundColor: 'rgba(0, 39, 142, 0.5)'
+                    },
+                    element,
+                    shouldTap: this.#settingsButtonSettings.shouldAutotap ?? false
+                }
+                : null
+        } else if (path === '/options') {
+            const element = await this.findExportElement()
+            return element != null
+                ? {
+                    style: {
+                        scale: 1.01,
+                        backgroundColor: 'rgba(0, 39, 142, 0.5)'
+                    },
+                    element,
+                    shouldTap: this.#exportButtonSettings.shouldAutotap ?? false
+                }
+                : null
+        } else if (path === '/intro') {
+            const element = await this.findSignInButton()
+            return element != null
```

Comment:
Do you mind if we use if statements instead of multi-line ternaries with complex objects.

Reply:
Wondering if it'd be not too bad if the object was simplified. Trying to avoid too much nesting here.